### PR TITLE
Add support for AWS Inferentia chips and Google TPUs & support latest PyTorch and TensorFlow versions

### PR DESCRIPTION
**Pull request by valeriosofi**
This PR adds the support of Inferentia chips and TPUs for PyTorch models.
Moreover, it adds support for PyTorch 2.0 and TensorFlow 2.12

### CODE REVIEW

File: apps/accelerate/speedster/docs/en/docs/installation.md [hunk 1]
```diff
@@ -169,14 +171,57 @@ If you want to manually install the requirements, this section collects links to
...
+
+## Set up Speedster on custom DL devices
+
+From version `0.10.0`, Speedster supports optimization of PyTorch models on `Google TPUs` and `AWS Inferentia` chips. 
+For these devices, the user must ensure that the required libraries are installed on the machine. 
+The following sections describe how to install the required libraries for each device.
+
+### Google TPUs
+
+In order to use a TPU, you must request a TPU-enabled VM from Google Cloud. You can consult the [official documentation](https://cloud.google.com/tpu/docs/run-calculation-pytorch?hl=en)
```
Comment (diegofiori) [inline apps/accelerate/speedster/docs/en/docs/installation.md:194]: I imagine here that we are not supporting tf on TPUs right?

File: apps/accelerate/speedster/docs/en/docs/installation.md [hunk 2]
```diff
@@ -169,14 +171,57 @@ If you want to manually install the requirements, this section collects links to
...
+
+You can find here the full guides to set up the EC2 instances and install the required libraries:
+- `inf1`: https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html#install-neuron-pytorch
+- `inf2`: https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/setup/pytorch-install.html#pytorch-neuronx-install
+
+After creating the EC2 instance and installing `torch_neuron`, you can follow these steps to set up Speedster:
+- Check that the `torch_neuron` library is installed, you can do this by running `python -c "import torch_neuron; print(torch_neuron.__version__)"` in the console (if using `inf1` instances, otherwise change `torch_neuron` with `torch_neuronx`);
+- Install Speedster by running `pip install speedster`. It's not required to install the deep learning compilers in this case, since they are not supported on AWS Inferentia.
+
+You are now ready to use Speedster on AWS Inferentia! Speedster will automatically detect the AWS Inferentia device and will use the `torch_neuron` library to optimize the model, comparing its performances with the original model running on the CPU.
```
Comment (diegofiori) [inline apps/accelerate/speedster/docs/en/docs/installation.md:227]: No support for TF?

File: nebullvm/operations/optimizations/base.py [hunk 3]
```diff
@@ -248,9 +255,7 @@ def _optimize(
                                     inference_learner,
                                     test_input_data,
                                     model_outputs,
-                                    metric_drop_ths
-                                    if q_type is not None
-                                    else CONSTRAINED_METRIC_DROP_THS,
+                                    metric_drop_ths,
```
Comment (diegofiori) [inline nebullvm/operations/optimizations/base.py:258]: why did you remove it?

File: nebullvm/tools/benchmark.py [hunk 4]
```diff
@@ -27,7 +27,11 @@
 
 
 def _get_dl_framework(model: Any):
-    if isinstance(model, torch.nn.Module) or str(model).startswith("Pytorch"):
+    if (
+        isinstance(model, torch.nn.Module)
+        or str(model).startswith("Pytorch")
+        or str(model).startswith("Torch")
+    ):
```
Comment (diegofiori) [inline nebullvm/tools/benchmark.py:34]: We should think about making naming conventions uniform

File: Dockerfile [hunk 5]
```diff
@@ -14,30 +14,13 @@ RUN apt-get install ffmpeg libsm6 libxext6  -y
 RUN apt-get install -y sudo wget
 
 # Install libraries
-RUN pip3 install --no-cache-dir torch torchvision --extra-index-url https://download.pytorch.org/whl/cu117  \
-    && python3 -m pip install --upgrade pip \
+RUN python3 -m pip install --upgrade pip \
+    && pip install --no-cache-dir torch torchvision --extra-index-url https://download.pytorch.org/whl/cu118  \
+    && pip install --no-cache-dir tensorflow \
     && pip install --no-cache-dir xformers \
-    && pip install --no-cache-dir -U diffusers \
```
[the remark this answers is not in the document]
Comment (valeriosofi) [reply Dockerfile:20]: It's not needed, it will be installed directly by speedster `auto_installer`

File: apps/accelerate/speedster/docs/en/docs/installation.md [hunk 6]
```diff
@@ -41,14 +43,14 @@ To customize the libraries installation you have two options:
...
-```python
+```bash
 python -m nebullvm.installers.auto_installer 
     --frameworks <frameworks> 
     --extra-backends <backends> 
     --compilers <compilers>
 ```
 
-!!! Description
+**Description**
```
[the remark this answers is not in the document]
Comment (valeriosofi) [reply apps/accelerate/speedster/docs/en/docs/installation.md:53]: This change must be reverted

File: apps/accelerate/speedster/docs/en/docs/installation.md [hunk 7]
```diff
@@ -133,12 +135,12 @@ The following table shows the supported combinations of frameworks, backends and
...
 | TensorFlow  | ONNX                      | TensorRT, OpenVINO                                                      |
 | ONNX        | /                         | TensorRT, OpenVINO                                                      |
-| HuggingFace | PyTorch, TensorFlow, ONNX | DeepSparse, TensorRT, Torch TensorRT, OpenVINO, Intel Neural Compressor |
+| Hugging Face | PyTorch, TensorFlow, ONNX | DeepSparse, TensorRT, Torch TensorRT, OpenVINO, Intel Neural Compressor |
 | Diffusers   | PyTorch, ONNX             | DeepSparse, TensorRT, Torch TensorRT, OpenVINO, Intel Neural Compressor |
 
 
-!!! info
-    Hugginface models can be of two types, PyTorch-based or TensorFlow-based. For PyTorch-based models, it is necessary to include `torch` as an extra-backend. For TensorFlow-based models, you must include `tensorflow` as an extra-backend.
+**info**
```
[the remark this answers is not in the document]
Comment (valeriosofi) [reply apps/accelerate/speedster/docs/en/docs/installation.md:142]: This change must be reverted

File: apps/accelerate/speedster/docs/en/docs/installation.md [hunk 8]
```diff
@@ -169,14 +171,57 @@ If you want to manually install the requirements, this section collects links to
...
 
     docker run -ti --gpus=all nebulydocker/nebullvm:latest
 
-After optimizing the model, you may decide to deploy it to production. Note that you need to have the deep learning compiler used to optimize the model and other components inside the production docker. For this reason, we have created several versions of the Docker nebullvm container in the [Docker Hub](https://hub.docker.com/repository/docker/nebulydocker/nebullvm), each containing only one compiler. Pull the image with the compiler that has optimized your model!
\ No newline at end of file
+After optimizing the model, you may decide to deploy it to production. Note that you need to have the deep learning compiler used to optimize the model and other components inside the production Docker. For this reason, we have created several versions of the Docker nebullvm container in the [Docker Hub](https://hub.docker.com/repository/docker/nebulydocker/nebullvm), each containing only one compiler. Pull the image with the compiler that has optimized your model!
+
+## Set up Speedster on custom DL devices
+
+From version `0.10.0`, Speedster supports optimization of PyTorch models on `Google TPUs` and `AWS Inferentia` chips.
```
[the remark this answers is not in the document]
Comment (valeriosofi) [reply apps/accelerate/speedster/docs/en/docs/installation.md:188]: 0.9.2? ;)

File: apps/accelerate/speedster/docs/en/docs/installation.md (hunk 1 above)
Comment (valeriosofi) [reply apps/accelerate/speedster/docs/en/docs/installation.md:194]: Yep right, I'll open an issue as soon as we perform the release
